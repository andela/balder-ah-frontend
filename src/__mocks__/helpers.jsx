import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render } from 'react-testing-library';

export const renderWithRouter = (
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {},
) => ({
  ...render(<Router history={history}>{ui}</Router>),
  history,
});

export const placeholder = {};
