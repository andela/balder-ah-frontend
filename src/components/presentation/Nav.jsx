import React from 'react';
import { Link } from 'react-router-dom';

const Nav = () => (
  <nav>
    <div className="nav-wrapper container">
      <Link to="/" className="brand-logo">
        AH Balder
      </Link>
      <ul id="nav-mobile" className="right hide-on-med-and-down">
        <li>
          <Link to="/login">Login</Link>
        </li>
        <li>
          <Link to="/singup">Sign Up</Link>
        </li>
      </ul>
    </div>
  </nav>
);

export default Nav;
