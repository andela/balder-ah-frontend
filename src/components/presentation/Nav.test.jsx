import React from 'react';
import { cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import { renderWithRouter } from '../../__mocks__/helpers';
import Nav from './Nav';

afterEach(cleanup);

describe('<Nav />', () => {
  test('should render correctly', () => {
    const { container } = renderWithRouter(<Nav />);
    expect(container).toMatchSnapshot();
  });
});
