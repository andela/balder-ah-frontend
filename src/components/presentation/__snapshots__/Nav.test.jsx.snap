// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render correctly 1`] = `
<div>
  <nav>
    <div
      class="nav-wrapper container"
    >
      <a
        class="brand-logo"
        href="/"
      >
        AH Balder
      </a>
      <ul
        class="right hide-on-med-and-down"
        id="nav-mobile"
      >
        <li>
          <a
            href="/login"
          >
            Login
          </a>
        </li>
        <li>
          <a
            href="/singup"
          >
            Sign Up
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
